src/osd/ReplicatedPG.cc

```diff
@@ -397,6 +397,7 @@ void ReplicatedPG::wait_for_unreadable_object(
     if (is_missing_object(soid)) {
       recover_missing(soid, v, cct->_conf->osd_client_op_priority, h);
     } else {
+      h->cache = true;
```

Comment:
cache = true here means that we should avoid caching the data?  Why?  We are in wait_for_unreadable_object, so there must be an IO pending.